web/app/mu-plugins/filters.php

```diff
@@ -235,3 +235,149 @@ function __rebuild_url_from_parts( array $parts ) : string {
 		( isset( $parts['fragment'] ) ? str_replace( '/', '', "#{$parts['fragment']}" ) : '' )
 	);
 }
+
+/**
+ * REST API Plain Permalink Fix
+ *
+ * Extracts the REST API endpoint from a potentially malformed path.
+ * Handles cases like /wp-json/v2/posts or /wp-json/wp/v2/posts.
+ *
+ * @since 1.2.3
+ * @param string $path The URL path component.
+ * @return string The extracted endpoint (e.g., /v2/posts) or '/'.
+ */
+function __extract_rest_endpoint( string $path ) : string {
+	$rest_route = '/'; // Default to base route
+	$wp_json_pos = strpos( $path, '/wp-json/' );
+
+	if ( $wp_json_pos !== false ) {
+		$extracted_route = substr( $path, $wp_json_pos + strlen( '/wp-json' ) ); // Get everything after /wp-json
+		// Special case: Handle the originally reported '/wp-json/wp/' malformation
+		if ( strpos( $extracted_route, 'wp/' ) === 0 ) {
+			$extracted_route = substr( $extracted_route, strlen( 'wp' ) ); // Remove the extra 'wp'
+		}
+		// Ensure the extracted route starts with a slash
+		if ( ! $extracted_route && $extracted_route[0] !== '/' ) {
+			$extracted_route = '/' . $extracted_route;
+		}
+		$rest_route = $extracted_route ?: '/'; // Use extracted route or default to base
+	}
+	return $rest_route;
```

Comment:
```suggestion
	if ( $wp_json_pos === false ) {
		return $rest_route;
	}
	$extracted_route = substr( $path, $wp_json_pos + strlen( '/wp-json' ) ); // Get everything after /wp-json
	// Special case: Handle the originally reported '/wp-json/wp/' malformation
	if ( strpos( $extracted_route, 'wp/' ) === 0 ) {
		$extracted_route = substr( $extracted_route, strlen( 'wp' ) ); // Remove the extra 'wp'
	}
	// Ensure the extracted route starts with a slash
	if ( ! $extracted_route && $extracted_route[0] !== '/' ) {
		$extracted_route = '/' . $extracted_route;
	}
	$rest_route = $extracted_route ?: '/'; // Use extracted route or default to base
```
disappointed. I thought you were on the return early train.